pkg/kapis/devops/v1alpha2/register.go

```diff
@@ -685,3 +695,25 @@ func AddS2IToWebService(webservice *restful.WebService, ksClient versioned.Inter
 	}
 	return nil
 }
+
+func AddJenkinsToContainer(webservice *restful.WebService, devopsClient devops.Interface, endpoint string) error {
+	if devopsClient == nil {
+		return nil
+	}
+	parse, err := url.Parse(endpoint)
+	if err != nil {
+		return err
+	}
+	parse.Path = strings.Trim(parse.Path, "/")
+	webservice.Route(webservice.GET("/jenkins/{path:*}").
+		To(func(request *restful.Request, response *restful.Response) {
+			u := request.Request.URL
+			u.Host = parse.Host
+			u.Scheme = parse.Scheme
+			jenkins.SetBasicBearTokenHeader(&request.Request.Header)
+			u.Path = strings.Replace(request.Request.URL.Path, fmt.Sprintf("/kapis/%s/%s/jenkins", GroupVersion.Group, GroupVersion.Version), "", 1)
+			httpProxy := proxy.NewUpgradeAwareHandler(u, http.DefaultTransport, false, false, nil)
```

Comment:
ErrorResponser can be useful for debugging, please don't pass nil

Reply:
fixed